pkg/controllers/utils/helpers.go

```diff
@@ -45,3 +48,43 @@ func GetAccountProviderType(account *v1alpha1.CloudProviderAccount) (v1alpha1.Cl
 		return "", fmt.Errorf("%s", ErrorMsgUnknownCloudProvider)
 	}
 }
+
+// UpdateIndexerObjects updates the objects in the indexer cache with the specified objects.
+func UpdateIndexerObjects(indexer cache.Indexer, objs interface{}) []error {
```

Comment:
As you have moved this to helper, where is AddIndexerObject?? Also, dont think its generic helper.. You expect object to be added as slice, so its custom implementation. Find a place in networkpolicy for this

Reply:
In indexer implementation, update and add are the same thing. In cloudRule indexer I'm also not using add at all, just update.